### PR TITLE
Hypervisor: Add support for default IPv6 gateway

### PR DESCRIPTION
**Pull request by asmithakarun**
This commit adds redfish support for patching
"IPv6DefaultGateway" property for each ethernet interfaces for a static configuration. In case of dynamic configuration, patching of this property will not be permitted in the backend networkd app and an error response will be sent back to the client.

Address the following defect:
https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=543926

The phosphor networkd changes are here:
https://github.com/ibm-openbmc/phosphor-networkd/pull/88

Tested By:

[1] PATCH -d '{"IPv6DefaultGateway":"<ipv6Gateway>"}' https://${bmc}/redfish/v1/Systems/hypervisor/EthernetInterfaces/eth<0/1>

Change-Id: Iaa48431147ba510eac6be9eec2f330303d588005

### CODE REVIEW

File: redfish-core/lib/hypervisor_system.hpp [hunk 1]
```diff
@@ -1237,13 +1258,15 @@ inline void requestRoutesHypervisorSystems(App& app)
...
+        std::optional<std::string> ipv6DefaultGateway;
 
         if (!json_util::readJsonPatch(
                 req, asyncResp->res, "HostName", hostName,
                 "IPv4StaticAddresses", ipv4StaticAddresses,
                 "IPv6StaticAddresses", ipv6StaticAddresses, "IPv4Addresses",
                 ipv4Addresses, "DHCPv4", dhcpv4, "DHCPv6", dhcpv6,
-                "StatelessAddressAutoConfig", statelessAddressAutoConfig))
+                "StatelessAddressAutoConfig", statelessAddressAutoConfig,
+                "IPv6DefaultGateway", ipv6DefaultGateway))
```
Comment (sunharis) [inline redfish-core/lib/hypervisor_system.hpp:1269]: This is readonly property. Can not patch.

For patching you must use IPv6StaticDefaultGateways property

**Review by sunharis — CHANGES_REQUESTED** [review]

File: redfish-core/lib/hypervisor_system.hpp (hunk 1 above)
Comment (asmithakarun) [reply redfish-core/lib/hypervisor_system.hpp]: Done

File: redfish-core/lib/hypervisor_system.hpp [hunk 2]
```diff
@@ -1416,6 +1439,31 @@ inline void requestRoutesHypervisorSystems(App& app)
...
+                    {
+                        defaultGwStr += "\"" + ipv6StaticDefaultGw[i] + "\"";
+                        if (i != ipv6StaticDefaultGw.size() - 1)
+                        {
+                            defaultGwStr += ",";
+                            continue;
+                        }
+                        defaultGwStr += "]}";
+                    }
+                    messages::propertyValueFormatError(
```
Comment (raviteja-b) [inline redfish-core/lib/hypervisor_system.hpp:1459]: say if user tries to configure multiple ipv6 default gateways, its correct as per schema,
so returning propertyValueFormatError does not Look right.
thinking if we can do other work around like take only first or last input entry or return a error description to say single entry supported

**Review by raviteja-b — APPROVED** [review]

File: redfish-core/lib/hypervisor_system.hpp (hunk 2 above)
Comment (asmithakarun) [reply redfish-core/lib/hypervisor_system.hpp]: >  its correct as per schema,
Done

**Review by sunharis — APPROVED** [review]

**Review by gtmills — APPROVED** [review]

**Pull request merged**